src/kernel/filesystem/fat32.zig

```diff
@@ -0,0 +1,1756 @@
+        Archive = 0x20,
+    };
+
+    ///
+    /// Get the short name from the entry. This will be encoded in UFT8. This will also include a
+    /// '.' between the name and extension and ignore trailing spaces.
+    ///
+    /// Arguments:
+    ///     IN self: *const ShortName - The short name entry to get the name from.
+    ///     IN buff: []u8             - A buffer to copy the UTF8 encoded name.
+    ///
+    /// Return: u32
+    ///     The index into the buffer where the last character was stored.
+    ///
+    pub fn getName(self: *const ShortName, buff: []u8) u32 {
+        var index: u8 = 0;
+        for (self.name) |char| {
+            if (char != ' ') {
+                // 0x05 is actually 0xE5
+                if (char == 0x05) {
+                    const utf16 = CodePage.toWideChar(.CP437, 0xE5);
+                    // The code page will return a valid character so the encode won't error
+                    index += std.unicode.utf8Encode(utf16, buff[index..]) catch unreachable;
+                } else {
+                    const utf16 = CodePage.toWideChar(.CP437, char);
+                    // The code page will return a valid character so the encode won't error
+                    index += std.unicode.utf8Encode(utf16, buff[index..]) catch unreachable;
+                }
+            } else {
+                break;
+            }
+        }
+        if (!self.isDir()) {
+            buff[index] = '.';
+            index += 1;
+            for (self.extension) |char| {
+                if (char != ' ') {
+                    const utf16 = CodePage.toWideChar(.CP437, char);
+                    // The code page will return a valid character so the encode won't error
+                    index += std.unicode.utf8Encode(utf16, buff[index..]) catch unreachable;
+                } else {
+                    break;
+                }
+            }
+        }
+        return index;
+    }
+
+    ///
+    /// Get the original short file name without encoding into UTF8.
+    ///
+    /// Arguments:
+    ///     IN self: *const ShortName - The short name entry to get the SFN name from.
+    ///
+    /// Return: [11]u8
+    ///     The original 11 characters of the short name entry.
+    ///
+    pub fn getSFNName(self: *const ShortName) [11]u8 {
+        var name: [11]u8 = [_]u8{' '} ** 11;
+        std.mem.copy(u8, name[0..], self.name[0..]);
+        std.mem.copy(u8, name[8..], self.extension[0..]);
+        return name;
+    }
+
+    ///
+    /// Check the attributes and check if the entry is a directory.
+    ///
+    /// Arguments:
+    ///     IN self: *const ShortName - The short name entry to get if this is a directory.
+    ///
+    /// Return: bool
+    ///     Whether the file is a directory or not.
+    ///
+    pub fn isDir(self: *const ShortName) bool {
+        return self.attributes & @enumToInt(ShortName.Attributes.Directory) == @enumToInt(ShortName.Attributes.Directory);
```

Comment:
`@enumToInt(ShortName.Attributes.Directory) == @enumToInt(ShortName.Attributes.Directory);` will never be false.

Reply:
`self.attributes & @enumToInt(ShortName.Attributes.Directory)` will evaluate first then the equality

Reply:
Oh of course, I was reading this after writing some C++ at work and read the `&` as `&&`! Perhaps some parentheses would be nice anyway.